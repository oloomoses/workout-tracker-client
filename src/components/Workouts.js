import React from 'react';
import { useSelector } from 'react-redux';
import { Container, Card } from 'react-bootstrap';
import Spinner from './Spinner';

const Workouts = () => {
  const loading = useSelector((state) => state.workout.loading);
  const workouts = useSelector((state) => state.workout.workouts);
  if (loading) {
    return <Spinner />;
  }

  return (
    <Container fluid>
      <div className="workouts-wrapper">
        { workouts.map((workout) => (
          <Card key={workout.id} className="workout-card-layout">
            <div className="d-flex justify-content-between">
              <div><img src={workout.icon} alt="icon" /></div>

              <div>
                <div>{workout.name}</div>
                <div className="trackIt-count text-center">13</div>
              </div>
            </div>
          </Card>
        ))}
      </div>
    </Container>
  );
};

export default Workouts;
