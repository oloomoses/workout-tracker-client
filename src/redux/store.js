import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import userRegistrationReducer from './user/userRegisterReducer';

const store = createStore(userRegistrationReducer, applyMiddleware(thunk));

export default store;
