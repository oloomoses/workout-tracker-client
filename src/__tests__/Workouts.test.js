import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Workouts from '../components/Workouts';
import store from '../redux/store';

describe('Workouts component', () => {
  it('Matches the snapshot', () => {
    const tree = renderer.create(<Provider store={store}><Workouts /></Provider>).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
