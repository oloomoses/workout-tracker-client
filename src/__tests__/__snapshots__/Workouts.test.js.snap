// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workouts component Matches the snapshot 1`] = `[Function]`;
